docs/guide/plugins.md

```diff
@@ -0,0 +1,202 @@
+
+- slot, in which we want to inject
+- vue file, which should be displayed there
+
+In the single plugin, we may inject in multiple plugin slots. Let's inject our `main.vue` file into `top-header-after` slot.
+
+```json{4,5}
+{
+  "slots": [
+    {
+      "name": "top-header-after",
+      "file": "main.vue"
+    }
+  ],
+  "settings": {}
+}
+```
+
+so our `main.vue` file can be as simple as
+
+```vue
+<template>
+  <div class="cool-promotion-plugin">
+    <h3>Hello from plugin!</h3>
+  </div>
+</template>
+<script>
+export default {
+  name: "CoolPromotionPlugin",
+  data() {
+    return {};
+  },
+};
+</script>
+<style lang="scss" scoped>
+.cool-promotion-plugin {
+  display: flex;
+  justify-content: center;
+  border: 1px solid green;
+}
+</style>
+```
+
+we run `shopware-pwa plugins`, and we see the effect
+
+![local-plugin-1](./../assets/local-plugin-1.png)
+
+Let's put some logic there, and we're getting the cheapest product and displaying it with link.
+
+```vue
+<template>
+  <div class="cool-promotion-plugin">
+    <!-- we can link to our product product -->
+    <nuxt-link
+      :to="getProductLink"
+      v-if="product"
+      class="cool-promotion-plugin__link"
+    >
+      <h3>{{ getName }} in special price! {{ getPrice | price }}</h3>
+    </nuxt-link>
+  </div>
+</template>
+<script>
+import { getProducts } from "@shopware-pwa/shopware-6-client";
+import {
+  getProductName,
+  getProductRegularPrice,
+  getProductUrl,
+} from "@shopware-pwa/helpers";
+
+export default {
+  name: "CoolPromotionPlugin",
+  data() {
+    return {
+      product: null,
+    };
+  },
+  async mounted() {
+    // let's get the cheapest product
+    const productsResult = await getProducts({
+      sort: {
+        field: "price",
+        desc: false,
+      },
+      pagination: {
+        limit: 1,
+      },
+    });
+    this.product = productsResult.data[0];
+  },
+  computed: {
+    getName() {
+      return getProductName({ product: this.product });
+    },
+    getProductLink() {
+      return getProductUrl(this.product);
+    },
+    getPrice() {
+      return getProductRegularPrice({ product: this.product });
+    },
+  },
+};
+</script>
+<style lang="scss" scoped>
+.cool-promotion-plugin {
+  display: flex;
+  justify-content: center;
+  border: 1px solid green;
+
+  &__link {
+    padding: 20px 5px;
+    min-height: 30px;
+    margin-right: 2rem;
+  }
+}
+</style>
+```
+
+so we achieved this:
+
+![local-plugin-2](./../assets/local-plugin-2.png)
+
+### Nesting plugins
+
+Okay, but what if we have two or more plugins for our `top-header-after` slot?
+Always the newest installed plugin in shopware instance is on top. But local plugins are over them. The easiest way is to picture that. On shopware instance, we have our HelloCody plugin, which displays random quotes. It injects into `top-header-after` slot as well. So we overrode that.
+
+:::warning Always provide a slot for other plugins
```

Comment:
How about the ordering of plugins? If there is no determined order, maybe point out, that the point of truth is in fact the installation date in SW?

Reply:
the line above is about the order
`Always the newest installed plugin in shopware instance is on top. But local plugins are over them. `
maybe somehow change that?

Reply:
https://github.com/DivanteLtd/shopware-pwa/pull/695/commits/bf2f4561abada74ea64364c83ecc4ec8028de7d8 is that any better? :)

Reply:
So I assume the order for multiple plugins that use the same slot is determined by the order in `local-plugins.json`? It's an edge case but at some point, people will ask and we might just include it as well.

Reply:
good point, added that as well :)
https://github.com/DivanteLtd/shopware-pwa/pull/695/commits/a5dbcaca65ad5f0bc70b37246b3026ddf86b6c0d